pkg/model/awsmodel/autoscalinggroup.go

```diff
@@ -101,6 +101,8 @@ func (b *AutoscalingGroupModelBuilder) buildLaunchTemplateTask(c *fi.ModelBuilde
 	if err != nil {
 		return nil, fmt.Errorf("error building cloud tags: %v", err)
 	}
+	// TODO: move this to CloudTagsForInstanceGroup when LaunchConfigurations are removed
+	tags["kubernetes.io/cluster/"+b.ClusterName()] = "owned"
```

Comment:
Perhaps call `b.CloudTags("", false)` ?

What would be the downside to moving that to CloudTagsFroInstanceGroup before LaunchConfigurations are removed? Why don't instances/volumes created by LaunchConfigurations throw the warning?

Reply:
> What would be the downside to moving that to CloudTagsFroInstanceGroup before LaunchConfigurations are removed?

I don't know if it would affect any current workflow if this tag is applied at ASG level. Launch templates are still experimental in Kops and shouldn't be an issue.

> Why don't instances/volumes created by LaunchConfigurations throw the warning?

Instances created with launch configurations don't tag root volumes at all. Because of this, the volumes are not tracked as part of the cluster by the delete function. They are still deleted automatically when the instance is deleted.

May also be a possibility to not track root volumes at all with the deleted function, even if they have the other tags.

Reply:
I like the idea of being consistent in what tags we apply.

I could see the `KubernetesCluster` tag being useful for cost control and leaving it out lacks consistency, so I think we should call `CloudTags` rather than setting `kubernetes.io/cluster/` directly.

For consistency, I'm in favor of moving things into `CloudTagsForInstanceGroup` sooner. Perhaps we should have `CloudTagsForInstanceGroup` call `CloudTags`. I don't know what the consequence of adding them to the template functions would be, though.

Reply:
I actually tried this as the first iteration, but the changes in tests were bigger than I expected.
I don't think should affect any workflow, but preferred not to change so much.